server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/rewriter/CobolLineIndicatorProcessorImpl.java

```diff
@@ -65,25 +68,31 @@ private CobolLine processLine(final CobolLine line) {
         result = processContinuationLine(line, conditionalRightTrimmedContentArea);
         break;
       case COMMENT:
-        result =
-            CobolLineUtils.copyCobolLineWithIndicatorAndContentArea(
-                COMMENT_TAG + WS, conditionalRightTrimmedContentArea, line);
-        break;
       case COMPILER_DIRECTIVE:
         result = CobolLineUtils.copyCobolLineWithIndicatorAndContentArea(WS, EMPTY_STRING, line);
         break;
       case NORMAL:
       case DEBUG:
       default:
-        result =
-            CobolLineUtils.copyCobolLineWithIndicatorAndContentArea(
-                WS, conditionalRightTrimmedContentArea, line);
+        result = processLineWithFloatingComment(line, conditionalRightTrimmedContentArea);
         break;
     }
 
     return result;
   }
 
+  private CobolLine processLineWithFloatingComment(
+      CobolLine line, String conditionalRightTrimmedContentArea) {
+    Matcher matchedLine = FLOATING_COMMENT_LINE.matcher(conditionalRightTrimmedContentArea);
+
+    if (matchedLine.matches()) {
```

Comment:
You can simplify it because it seems that the matcher will match any line.

Reply:
Somehow if  do not call Matcher.matches() before I try to extract the group value, I see "java.lang.IllegalStateException: No match found" error.

Reply:
You are right, you need to call `matches()` to do the actual matching. But result will be `true` in any way.
I've found this example: https://github.com/eclipse/che-che4z-lsp-for-cobol/blob/development/server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/reader/CobolLineReaderImpl.java#L112
For some reason, I describe `else` branch. Maybe because Coverity found it suspicious.

Reply:
Seems it's not needed and will confuse more.